src/components/CheckIn/util.js

```diff
@@ -41,4 +46,12 @@ const checkInItems = (pieces, mutator) => {
   });
 };
 
-export { checkInItems as default };
+export const getMixedPieceAndItem = (pieces, itemsMap) => {
```

Comment:
I was thinking to have such function only for 1 piece. In your way you create arrays for all items several times. But if you have such function for 1 piece you have to do it only once

Reply:
Ok, updated